bosh/plugin.go

```diff
@@ -4,27 +4,64 @@ import (
 	"encoding/json"
 	"log"
 	"net/http"
+
+	"github.com/cloudfoundry/bosh-agent/jobsupervisor/monit"
 )
 
 type plugin struct {
-	report report
+	report          report
+	monitHTTPClient monit.Client
 }
 
-func NewPlugin(hostname, jobSpecPath string) *plugin {
+func NewPlugin(hostname, jobSpecPath, platform string) (*plugin, error) {
 	report := newReport(hostname)
 
 	if jobSpec, err := loadSpec(jobSpecPath); err != nil {
 		log.Printf("error loading job spec from %q: %v\n", jobSpecPath, err)
+		return nil, err
 	} else {
-		report.AddJob(jobSpec)
+		report.SetInstanceSpecData(jobSpec)
+	}
+
+	var monitHTTPClient monit.Client
+	var err error
+
+	if platform == "ubuntu" {
```

Comment:
Some questions w/r to `platform`:
1. Is there a reason you chose `ubuntu` over `linux`? I assume one would still specify `ubuntu` on the command line even though they run on a CentOS stemcell?
2. If this is more about Linux in general, we could actually use Go build tags to distinguish between the platform at build time, i.e. have a functional implementation of `monit.Client` that specifies `+build linux darwin` and a noop client that uses `+build windows`. That way we could get rid of the command line argument. Just an idea.
3. Out of interest, does the Bosh Agent on Windows not use Monit at all or are you more concerned about the location of the credentials file etc?